include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
and/or also test for `__cplusplus`?  For example, both GCC and Clang provide `std::byte` when compiling with `-std=c++1z` which also implies a value of `__cplusplus` over 201703L.